src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Extensions/StringExtensionTests.cs

```diff
@@ -5,129 +5,158 @@ namespace GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions
-        {
-            Assert.Equal("fooBar", "foo_bar".CamelCase());
-            Assert.Equal("fooBar", "foo_Bar".CamelCase());
-            Assert.Equal("fooBar", "Foo_Bar".CamelCase());
-            Assert.Equal("fOOBar", "FOO_Bar".CamelCase());
-        }
-
-        [Fact]
-        public void NullStringCanBePascalCased()
-        {
-            string input = null;
-            Assert.Null(input.PascalCase());
-        }
-
-        [Fact]
-        public void EmptyStringCanBePascalCased()
-        {
-            Assert.Equal(string.Empty, string.Empty.PascalCase());
-        }
-
-        [Fact]
-        public void ValuesAreNotTouchedByPascalCase()
-        {
-            Assert.Equal("Foo", "Foo".PascalCase());
-            Assert.Equal("FOO", "FOO".PascalCase());
-        }
-
-        [Fact]
-        public void PascalCasedStringRemainsPascalCased()
-        {
-            Assert.Equal("PascalCase", "PascalCase".PascalCase());
-        }
-
-        [Fact]
-        public void CamelCaseStringIsPascalCased()
-        {
-            Assert.Equal("FooBar", "fooBar".PascalCase());
-            Assert.Equal("FooBAR", "fooBAR".PascalCase());
-        }
-
-        [Fact]
-        public void UnderscoreSeparatedStringsArePascalCased()
-        {
-            Assert.Equal("FooBar", "foo_bar".PascalCase());
-            Assert.Equal("FooBar", "foo_Bar".PascalCase());
-            Assert.Equal("FooBar", "Foo_Bar".PascalCase());
-            Assert.Equal("FOOBar", "FOO_Bar".PascalCase());
-        }
-
-        [Fact]
-        public void NullStringsCanBeScreamingSnakeCased()
-        {
-            string input = null;
-            Assert.Null(input.ScreamingSnakeCase());
-        }
-
-        [Fact]
-        public void EmptyStringCanBeScreamingSnakeCased()
-        {
-            Assert.Equal(string.Empty, string.Empty.ScreamingSnakeCase());
-        }
-
-        [Fact]
-        public void AcronymsAreAlreadyScreamingSnakeCased()
-        {
-            Assert.Equal("ABC", "ABC".ScreamingSnakeCase());
-        }
-
-        [Fact]
-        public void CamelCasedStringsAreScreamingSnakeCased()
-        {
-            Assert.Equal("FOO", "foo".ScreamingSnakeCase());
-            Assert.Equal("CAMEL_CASE_STRING", "camelCaseString".ScreamingSnakeCase());
-        }
-
-        [Fact]
-        public void UnderscoreSeparatedStringsAreScreamingSnakeCased()
-        {
-            Assert.Equal("FOO_BAR", "foo_bar".ScreamingSnakeCase());
-            Assert.Equal("FOO_BAR", "foo_Bar".ScreamingSnakeCase());
-            Assert.Equal("FOO_BAR", "Foo_Bar".ScreamingSnakeCase());
-            Assert.Equal("FOO_BAR", "FOO_Bar".ScreamingSnakeCase());
-        }
-        
-        [Fact]
-        public void PascalCasedStringsAreScreamingSnakeCased()
-        {
-            Assert.Equal("FOO_BAR", "FooBar".ScreamingSnakeCase());
-            Assert.Equal("FOO_BAR", "FooBAr".ScreamingSnakeCase());
+        public class CamelCaseTests
```

Comment:
Something seems off with the formatting here so it's difficult to see what's changed.

Reply:
Don't think there's anything wrong with the formatting. I think this is just how the diff looks because I've wrapped the blocks of tests as nested classes to make them easier to distinguish from each other.

I've added the following cases:

```csharp
// For CamelCaseTests
[Fact]
public void StringsWithNonAlphaNumericCharactersAreCamelCased()
{
    Assert.Equal("fooBarBaz", "foo bar  baz".CamelCase());
    Assert.Equal("fooBarBaz", "foo-bar--baz".CamelCase());
    Assert.Equal("fooBarBaz", "foo.bar..baz".CamelCase());
    Assert.Equal("fooBarBaz", "foo_bar__baz".CamelCase());
    Assert.Equal("fooBarBaz", "foo,bar,,baz".CamelCase());
}

// For PascalCaseTests
[Fact]
public void StringsWithNonAlphaNumericCharactersArePascalCased()
{
    Assert.Equal("FooBarBaz", "foo bar  baz".PascalCase());
    Assert.Equal("FooBarBaz", "foo-bar--baz".PascalCase());
    Assert.Equal("FooBarBaz", "foo.bar..baz".PascalCase());
    Assert.Equal("FooBarBaz", "foo_bar__baz".PascalCase());
    Assert.Equal("FooBarBaz", "foo,bar,,baz".PascalCase());
}
```